### PR TITLE
Addition of BluFor Enhanced Mechanized Company Template

### PR DESCRIPTION
**Pull request by Alablm**
Adds a new "faction" using potato_usmc units to create a new company composition making use of the updated CSW units while maintaining MMG and MAT support inside the platoon structure. Provides 166 slots not including JIPs. Contains 8 loadouts modified to fit the structure. Squads are designed to fit in most of our available BluFor APC/IFVs without modification, but have two spare rifleman that can be cut to allow squeezing into the smaller vehicles.

### CODE REVIEW

**Comment by Alex6511** [comment]
Just by the way, the reason why I segregated MSV to redfor when I remade it, and had planned for marines/airborne to go in the other factions was because there is a maximum limit of **groups** per mission in Arma 3. The number is high, but if you end up implementing Marines/Airborne as well you are probably starting to get close.

**Comment by Alablm** [comment]
Mission.sqm has been De-binerized, along with some minor fixes.

Also with regards to the group limit, the limit is 288 _per side_, meaning we could support 7 different factions equaling the default 41 groups in BWMF on each side before hitting the cap.

**Comment by Alex6511** [comment]
Fair enough, I'll look at the mission.sqm next but it will obviously take awhile given the file size.

@PabstMirror The group limit doesn't seem to be a problem, but I would still be concerned about the in-game performance implications when loading the template blank in eden (FPS was already pretty low, if memory serves) and the actual size of the mission.sqm in version control. Leave that up to you though.

**Comment by Alablm** [comment]
The template is not a direct rip of MSV for the record, it has a different platoon and company structure then both our base templates. The only reason I'm describing it as MSV is because its intended to be used in a mechanised setting.

**Comment by Alex6511** [comment]
Oh, in that case it might be a good idea to modify the name to something like CAB (Combined Arms Batallion)

**Comment by Alablm** [comment]
Updated group names to fit with Alex's suggestion of a separate name to avoid confusion.

File: CfgLoadouts.hpp [hunk 1]
```diff
@@ -69,10 +71,18 @@ class CfgLoadouts {
   // US: HK416 - Multi-Terrain Pattern "Loadouts\us_hk416_mtp_apex.hpp" (apex)
   // US: HK416 - Tropical "Loadouts\us_hk416_trop_apex.hpp" (apex)
   // US: M4 - Gray/Green "Loadouts\us_m4_ucp.hpp"
+  // US: M4 - Gray/Green "Loadouts\blumsv_us_m4_ucp.hpp" (msv)
   // US: M4 - Tan "Loadouts\us_m4_ocp.hpp"
+  // US: M4 - Tan "Loadouts\blumsv_us_m4_ocp.hpp" (msv)
   // US: M16 - M81 "Loadouts\us_m16_m81.hpp"
+  // US: M16 - Marine Pattern Desert "Loadouts\usmc_m16_marpatd.hpp"
```
Comment (Alex6511) [inline CfgLoadouts.hpp:78]: Is this supposed to be blumsv?

File: CfgLoadouts.hpp [hunk 2]
```diff
@@ -69,10 +71,18 @@ class CfgLoadouts {
...
+  // US: M16 - Marine Pattern Desert "Loadouts\blumsv_usmc_m16_marpatd.hpp" (msv)
+  // US: M16 - Marine Pattern Woodland "Loadouts\usmc_m16_marpatw.hpp"
```
Comment (Alex6511) [inline CfgLoadouts.hpp:80]: Same here, if not what is it?

If it's for "normal" loadouts (I.E. TvT loadouts) I think it should be blu_

File: CfgLoadouts.hpp [hunk 3]
```diff
@@ -95,10 +108,9 @@ class CfgLoadouts {
     #include "Loadouts\csat_sama_hex.hpp"
   };
 
-  class potato_msv { // MSV
+  class potato_msv { // Opfor MSV
     #include "Loadouts\msv_csat_sama_hex.hpp"
   };
-
```
Comment (Alex6511) [inline CfgLoadouts.hpp:101]: This space should still be here I think, make sure it matches the formatting on the other sections

File: CfgLoadouts.hpp [hunk 4]
```diff
@@ -83,6 +93,9 @@ class CfgLoadouts {
     #include "Loadouts\us_mx_mtp.hpp"
   };
 
+  class potato_usmc { // Blufor MSV
+    #include "Loadouts\blumsv_us_mx_mtp.hpp"
+  };
```
Comment (Alex6511) [inline CfgLoadouts.hpp:98]: Add a space here, make sure it matches the formatting of the other sections.

File: Loadouts/blumsv_brit_l85_ddpm.hpp [hunk 5]
```diff
@@ -0,0 +1,318 @@
+// Author: BWMF
+// Description: US: HK416 - Marine Pattern Desert
+
+#include "undef.hpp" // Reset defines
+
+// Rifle
+#define RIFLE "CUP_arifle_L85A2_G","CUP_arifle_L85A2_NG"
+#define RIFLE_MAG "30Rnd_556x45_Stanag_red:8","30Rnd_556x45_Stanag_Tracer_Red:2"
+#define RIFLE_ATTACHMENTS "CUP_acc_ANPEQ_2"
+#define ALT_OPTICS STANAG_OPTICS
```
Comment (Alex6511) [inline Loadouts/blumsv_brit_l85_ddpm.hpp:10]: Oh my god did we really define this as STANG_OPTICS? I'll have to fix that at some point.

File: Loadouts/blumsv_brit_l85_ddpm.hpp [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+#define MAT_OPTIC "CUP_optic_SMAW_Scope"
+//HMG
+#define HMG "ace_cswCompatCUP_cup_m2"
+#define HMG_TRI "ace_csw_m3CarryTripod"
+#define HMG_TRI_LOW "ace_csw_m3CarryTripodLow"
+#define HMG_MAG "ace_csw_100Rnd_127x99_mag:3"
+//HAT
+#define HAT "ace_cswCompatCUP_cup_tow"
+#define HAT_TRI "ace_csw_m220CarryTripod"
+#define HAT_TRI_LOW ""
```
Comment (Alex6511) [inline Loadouts/blumsv_brit_l85_ddpm.hpp:41]: @TheCandianVendingMachine Is this right?

File: Loadouts/blumsv_brit_l85_ddpm.hpp [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+  items[] += {RADIO_MR};
+};
+class potato_usmc_coy: potato_usmc_sl {// CO
+  items[] += {RADIO_LR};
+};
+class potato_usmc_xo: potato_usmc_coy {// XO/PL Sgt
+};
+class potato_usmc_sgt: potato_usmc_rifleman {
+  backpack[] = {SIDE_UAV_BACKPACK};
+  linkedItems[] += {SIDE_UAV_TERMINAL};
```
Comment (Alex6511) [inline Loadouts/blumsv_brit_l85_ddpm.hpp:125]: Is this typical? To have a UAV terminal? I can't remember the pattern.

File: Loadouts/blumsv_brit_l85_mtp.hpp [hunk 8]
```diff
@@ -0,0 +1,317 @@
+// Author: BWMF
+// Description: US: HK416 - Marine Pattern Desert
```
Comment (Alex6511) [inline Loadouts/blumsv_brit_l85_mtp.hpp:2]: This is wrong,

Also you're the author.

File: Loadouts/blumsv_brit_l85_ddpm.hpp [hunk 9]
```diff
@@ -0,0 +1,318 @@
+// Author: BWMF
+// Description: US: HK416 - Marine Pattern Desert
```
Comment (Alex6511) [inline Loadouts/blumsv_brit_l85_ddpm.hpp:2]: This is wrong, also you're the author.

File: Loadouts/blumsv_us_m4_ocp.hpp [hunk 10]
```diff
@@ -0,0 +1,312 @@
+// Author: BWMF
+// Description: US: HK416 - Marine Pattern Desert
```
Comment (Alex6511) [inline Loadouts/blumsv_us_m4_ocp.hpp:2]: See prev

File: Loadouts/blumsv_us_mx_mtp.hpp [hunk 11]
```diff
@@ -0,0 +1,309 @@
+// Author: BWMF
+// Description: US: HK416 - Marine Pattern Desert
```
Comment (Alex6511) [inline Loadouts/blumsv_us_mx_mtp.hpp:2]: See prev

File: Loadouts/blumsv_us_mx_trop_apex.hpp [hunk 12]
```diff
@@ -0,0 +1,312 @@
+// Author: BWMF
+// Description: US: HK416 - Marine Pattern Desert
```
Comment (Alex6511) [inline Loadouts/blumsv_us_mx_trop_apex.hpp:2]: See prev

File: Loadouts/blumsv_usmc_m16_marpatd.hpp [hunk 13]
```diff
@@ -0,0 +1,309 @@
+// Author: BWMF
+// Description: US: M16 - Marine Pattern Desert
```
Comment (Alex6511) [inline Loadouts/blumsv_usmc_m16_marpatd.hpp:2]: See prev

File: Loadouts/blumsv_usmc_m16_marpatw.hpp [hunk 14]
```diff
@@ -0,0 +1,309 @@
+// Author: BWMF
+// Description: US: M16 - Marine Pattern Woodland
```
Comment (Alex6511) [inline Loadouts/blumsv_usmc_m16_marpatw.hpp:2]: Se prev

**Comment by Alex6511** [comment]
Here's some examples of things that should be cleaned out, yours is on the right, master is on the left:
image
image
image
image

A lot of this might not seem like it matters, but your mission.sqm is nearly 300 KB larger than the one in master, it all adds up.

The angles and elevation offsets at the very least can be easily removed, the more minor stuff (like tons of uneeded modules) Is a lot harder to track down, so without AACOs attention to detail we might just have to let some of that go.

File: CfgLoadouts.hpp (hunk 1 above)
Comment (Alablm) [reply CfgLoadouts.hpp:78]: No, that's for the standard loadout. The EMC version is on line 78. Going to be changing all the paths from blumsv to emc right now.

File: CfgLoadouts.hpp (hunk 2 above)
Comment (Alablm) [reply CfgLoadouts.hpp:80]: Normal loadouts are all led by country identifiers, ex Canadian Forces are can_ russians are ru_ ect. The USMC loadouts are from the base framework.

File: Loadouts/blumsv_brit_l85_ddpm.hpp (hunk 7 above)
Comment (Alablm) [reply Loadouts/blumsv_brit_l85_ddpm.hpp:125]: This is the COY UAV operator, so yes. Potato: USMC doesnt have a UAV operator unit, so I had to make do. I've added a comment to that effect.

File: Loadouts/blumsv_brit_l85_ddpm.hpp (hunk 6 above)
Comment (Alablm) [reply Loadouts/blumsv_brit_l85_ddpm.hpp:41]: If you're asking if they're the right items, yes https://puu.sh/FJ7w7/ca11f628fe.jpg

**Comment by Alablm** [comment]
Positional changes puushes died, these are the same pictures
image
image

**Comment by Alablm** [comment]
Judging by the recent uptick in Blufor mechanised infantry missions, there is an obvious want in the group for this template. What kind of testing are you guys looking for to integrate this PR?

**Comment by Alablm** [comment]
Closing this PR

**Pull request closed without merging**